### PR TITLE
chore: Update model doc for code_llama

### PR DESCRIPTION
**Pull request by AbhishekRP2002**
# What does this PR do?
aims to handle https://github.com/huggingface/transformers/issues/36979#issuecomment-2758560598

sub part of https://github.com/huggingface/transformers/issues/36979

([36979](https://github.com/huggingface/transformers/issues/36979))

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/code_llama.md [hunk 1]
```diff
@@ -14,107 +14,158 @@ rendered properly in your Markdown viewer.
...
+```py
+import torch
+from transformers import pipeline
 
-Check out all Code Llama model checkpoints [here](https://huggingface.co/models?search=code_llama) and the officially released ones in the [Meta Llama org](https://huggingface.co/meta-llama).
+pipe = pipeline(
+    "text-generation",
+    model="meta-llama/CodeLlama-7b-hf",
+    torch_dtype=torch.float16,
+    device_map="auto"
```
Comment (stevhliu) [inline docs/source/en/model_doc/code_llama.md:47]:
```suggestion
    device=0
```

File: docs/source/en/model_doc/code_llama.md [hunk 2]
```diff
@@ -14,107 +14,158 @@ rendered properly in your Markdown viewer.
...
-```python
->>> from transformers import pipeline
->>> import torch
-
->>> generator = pipeline("text-generation",model="meta-llama/CodeLlama-7b-hf",torch_dtype=torch.float16, device_map="auto")
->>> generator('def remove_non_ascii(s: str) -> str:\n    """ <FILL_ME>\n    return result', max_new_tokens = 128)
-[{'generated_text': 'def remove_non_ascii(s: str) -> str:\n    """ <FILL_ME>\n    return resultRemove non-ASCII characters from a string. """\n    result = ""\n    for c in s:\n        if ord(c) < 128:\n            result += c'}]
-```
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/codellama-attn-mask.png"/>
```
Comment (stevhliu) [inline docs/source/en/model_doc/code_llama.md:154]: Can you open a PR on https://huggingface.co/datasets/huggingface/documentation-images/tree/main/transformers/model_doc to upload the image here?

**Review by stevhliu — COMMENTED** [review]
Great job! 👏

File: docs/source/en/model_doc/code_llama.md (hunk 2 above)
Comment (AbhishekRP2002) [reply docs/source/en/model_doc/code_llama.md:143]: https://huggingface.co/datasets/huggingface/documentation-images/discussions/472
hi @stevhliu, i have opened a PR, can u pls verify if everything looks good or if there's anything else i need to consider ?

Comment (AbhishekRP2002) [reply docs/source/en/model_doc/code_llama.md:143]: https://huggingface.co/datasets/huggingface/documentation-images/discussions/472
hi @stevhliu, i have opened a PR, can u pls verify if everything looks good or if there's anything else i need to consider ?

**Review by stevhliu — COMMENTED** [review]
Nice, just a few more comments :)

File: docs/source/en/model_doc/code_llama.md [hunk 3]
```diff
@@ -14,108 +14,142 @@ rendered properly in your Markdown viewer.
...
+</div>
 
-Code Llama has the same architecture as the `Llama2` models, refer to [Llama2's documentation page](llama2) for the API reference.
-Find Code Llama tokenizer reference below. 
-</Tip>
+## Notes
 
+- Infilling is only available in the 7B and 13B base models, and not in the Python, Instruct, 34B, or 70B models.
+- Use the `<FILL_ME>` token where you want your input to be filled. The tokenizer splits this token to create a formatted input string that follows the [original training pattern](https://github.com/facebookresearch/codellama/blob/cb51c14ec761370ba2e2bc351374a79265d0465e/llama/generation.py#L402). This is more robust than preparing the pattern yourself.
```
[the remark this answers is not in the document]
Comment (AbhishekRP2002) [reply docs/source/en/model_doc/code_llama.md:146]: Implemented the suggested changes ! Thank you for assisting. @stevhliu

[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/source/en/model_doc/code_llama.md:146]: The code example here isn't indented yet. It should be formatted like below:

```md
- some text
   
   ```py
   code snippet
   ```

File: docs/source/en/model_doc/code_llama.md [hunk 4]
```diff
@@ -14,108 +14,154 @@ rendered properly in your Markdown viewer.
...
+import torch
+from transformers import AutoModelForCausalLM, CodeLlamaTokenizer, BitsAndBytesConfig
 
-Tips:
-- The infilling task is supported out of the box. You should be using the `tokenizer.fill_token` where you want your input to be filled.
-- The model conversion script is the same as for the `Llama2` family:
+bnb_config = BitsAndBytesConfig(load_in_4bit=True, bnb_4bit_compute_dtype=torch.bfloat16, bnb_4bit_quant_type="nf4", bnb_4bit_use_double_quant=True)
+tokenizer = CodeLlamaTokenizer.from_pretrained("meta-llama/CodeLlama-34b-hf")
+model = AutoModelForCausalLM.from_pretrained(
+   "meta-llama/CodeLlama-34b-hf,
```
Comment (stevhliu) [inline docs/source/en/model_doc/code_llama.md:116]:
```suggestion
   "meta-llama/CodeLlama-34b-hf",
```

**Review by stevhliu — APPROVED** [review]
Super close! Remember to mark as ready for review when you're done 🤗

**Comment by AbhishekRP2002** [comment]
> Super close! Remember to mark as ready for review when you're done 🤗

Thank you ! marked the PR as Ready
